bin/picom-trans

```diff
@@ -46,194 +49,278 @@ print_usage() {
-window=
-opacity=
-cur=
-action=
-treeout=
-wid=
-topmost=
-lineno=
-option=
-v=
-
-# We make getopts stop on any argument it doesn't recognize
-# or errors on. This allows for things like `picom-trans -5`
-# as well as `picom-trans -c +5 -s` (contrived example).
-while test $# -gt 0; do
-  # Reset option index
-  OPTIND=1
-
-  # Read options
-  while getopts ':hscrtdgn:w:o:-:' option "$@"; do
-    if test "$option" = '-'; then
-      case "$OPTARG" in
-        help | select | current | reset | toggle | delete | get)
-          v=''
-        ;;
-        name | window | opacity)
-          eval v=\$$OPTIND
-          OPTIND=$((OPTIND + 1))
-        ;;
-        name=* | window=* | opacity=*)
-          v=$(echo "$OPTARG" | sed -E 's/^[^=]+=//')
-        ;;
-        *)
-          echo "$0: illegal option $OPTARG" >& 2
-          exit 1
-        ;;
-      esac
-      option=$(echo "$OPTARG" | cut -c 1)
-      OPTARG=$v
-    fi
-    case "$option" in
-      h) print_usage; exit 0 ;;
-      s) wprefix=''; window='' ;;
-      c)
-        active=$(xprop -root -notype _NET_ACTIVE_WINDOW \
-          | grep -Eo '0x[[:xdigit:]]+' | head -n 1)
-        wprefix='-id'; window=$active
-      ;;
-      r) action='reset' ;;
-      t) action='toggle' ;;
-      d) action='delete' ;;
-      g) action='get' ;;
-      n) wprefix='-name'; window=$OPTARG ;;
-      w) wprefix='-id'; window=$OPTARG ;;
-      o) opacity=$OPTARG ;;
-      \?) break ;;
+    # Separate  "--ARG=VAL"  into  "--ARG" "VAL"
+    case "$1" in
+      (--opacity=*|--name=*|--window=*)
+        ARG="$(echo "$1" | sed -E 's/(--[^=]+)=.*$/\1/')"
+        VAL="${1##${ARG}=}"
+        shift && set -- "$ARG" "$VAL" "$@"
     esac
-  done
 
-  # Read positional arguments
-  shift $((OPTIND - 1))
-  test -n "$1" && opacity=$1 && shift
-done
+    # Turn into short form
+    case "$1" in
+      (--help|--opacity|--get|--delete|--toggle|--reset|--select|--current|--name|--window)
+        ARG=${1#-}                      # remove one '-' from prefix
+        ARG="$(echo $ARG | cut -c -2)"  # get first two characters
+        shift && set -- "$ARG" "$@"
+    esac
 
-# clean up opacity. xargs == a poor man's trim.
-opacity=$(echo "$opacity" | xargs | sed 's/%//g')
+    # Prepare for next iteration
+    ARG="$1"
+    shift && set -- "$@" "$ARG"
+    i=$(( i + 1 ))
+  done
 
-# Validate opacity value
-if test -z "$action" && ! echo "$opacity" | grep -qE '^[+-]?[0-9]+$'; then
-  echo "Invalid opacity specified: $opacity."
-  exit 1
-fi
 
-# Reset opacity for all windows
-if test x"$action" = x'reset'; then
-  xwininfo -root -tree \
-  | sed -n 's/^     \(0x[[:xdigit:]]*\).*/\1/p' \
-  | while IFS=$(printf '\n') read wid; do
-    xprop -id "$wid" -remove _NET_WM_WINDOW_OPACITY
+  while test $# -gt 0; do
+    # 15 +15 -15 are opacity values
+    # filter them out here, otherwise getopts will complain
+    if echo "$1" | grep -qE '^[+-]?[[:digit:]]+$'; then
+      target_opacity=$1
+      shift
+      continue
+    fi
```

Comment:
Seems to work for `picom-trans -c +10` but not for `picom-trans -c -10` (for example). It seems `getopts` happily accepts `-10` as an argument before we have a chance to catch it here.

This might work however, if we let `getopts` handle exactly one OPTION (with its OPTARG) at a time.

Reply:
Right. This issue has been troubling me ever since I started writing this.

Do I use the workaround in the current version? i.e 
```
grep -qE 's/^-%([[:digit:]]+)%?$/~\1/;s/^+%([[:digit:]]+)%?$/#\1/;'
```
To replace `-15` into `~15` and `+15` into `#15`

Reply:
I think I will have to re-write the getopts loop.

This will take some time.

Reply:
0f4283a should fix this.

Reply:
0f4283 has been overridden by 83c215a

Please check if 83c215a solves this issue.